### PR TITLE
Extract duplicated code & remove unused parameter in DataSetTasksListGridViewImpl

### PR DESCRIPTION
**Pull request by jhrcek**
Extracted 4x repeated task list filter tab setup code to 2 private methods. A lot can be done to de-duplicate the code in this class, but I'd rather proceed in smaller steps.

Also removed unused parameter "String role " from each of the 4 methods.

### CODE REVIEW

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridViewImpl.java [hunk 1]
```diff
@@ -570,280 +570,150 @@ private void initGroupTabFilter( GridGlobalPreferences preferences,
...
         builder.initBuilder();
 
         builder.dataset(HUMAN_TASKS_WITH_USER_DATASET);
-        List<Comparable> names = new ArrayList<Comparable>();
-
-        for ( String s : states ) {
-            names.add( s );
-        }
+        List<Comparable> names = asComparables(states);
         builder.filter( COLUMN_STATUS, equalsTo( COLUMN_STATUS, names ) );
```
Comment (cristianonicolai) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridViewImpl.java]: you probably don't need the `asComparables` method, if you do `new ArrayList<>(states)` in here.

**Comment by cristianonicolai** [comment]
@nmirasch would you mind reviewing this change?

File: jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridViewImpl.java (hunk 1 above)
Comment (jhrcek) [inline jbpm-console-ng-human-tasks/jbpm-console-ng-human-tasks-client/src/main/java/org/jbpm/console/ng/ht/client/editors/taskslist/grid/dash/DataSetTasksListGridViewImpl.java]: Good point. Let me review that. I'll add note when it's ready. @nmirasch Please wait with the review until I implement Cristiano's suggestion.

**Comment by nmirasch** [comment]
@cristianonicolai @jhrcek reviewed, it's good for me.

**Comment by jhrcek** [comment]
As suggested I replaced 

    List<Comparable> names = asComparables(states);

by

    List<Comparable> names = new ArrayList<>(states);

Ready for review/merge.

**Comment by nmirasch** [comment]
+1 @jhrcek Thanks!

**Pull request merged**